File: Java/javaperformancetester/src/main/java/edu/imepac/javaperformancetester/dtos/ResponseDto.java
package edu.imepac.javaperformancetester.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/*ATENÇÃO!!!:
    Essa é a nova classe que os controllers devem retornar.
    Olhem no controller dos vagões como eu usei
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
public class ResponseDto<T> {//<T> pra poder colocar qualquer tipo de dado nela.
    private long tempoCpuMs;
    private T dados;//aqui q vai ficar os dados de vagão e encoste (depende do controller sendo usando no momento)
}

File: Kotlin/kotlinperformancetester/src/main/kotlin/edu/imepac/kotlinperformancetester/dtos/ResponseDto.kt
package edu.imepac.kotlinperformancetester.dtos

import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
data class ResponseDto<T>(
    val cpuTimeMs: Long,
    val data: T
){
}
